### PR TITLE
LF: clean scala Decoder test.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
As requested by Martin in #8069, we extract refactoring of 
DecodeV1Spec in a separate PR.

This advances the state of #7155

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala [hunk 1]
```diff
@@ -45,12 +45,18 @@ class DecodeV1Spec
   val dummyModuleDName = DamlLf1.DottedName.newBuilder().addSegments(dummyModuleStr).build()
   val dummyModuleName = DottedName.assertFromString(dummyModuleStr)
 
+  import com.daml.lf.transaction.VersionTimeline.Implicits._
+
+  val lfVersions =
```
Comment (hurryabit) [inline daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala:50]: What do you think about adding two functions of types
```scala
forEveryBefore, forEveryAtOrAfter: (Version) => (Version => Unit) => Unit
```
Then we could simply swap out the `forEvery`s below and avoid a lot of `if`s.

**Review by hurryabit — APPROVED** [review]
This is great. I was seriously wondering why the tests are the way they were. Thanks for improving this!

**Comment by remyhaemmerle-da** [comment]
blocked by #8069

File: daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala (hunk 1 above)
Comment (hurryabit) [reply daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala]: Thanks for doing this! 🙇

**Pull request merged**